File: passwords-manager/src/models/application.ts
import {ApplicationType} from "./application-type";

export class Application {
  idApplication: number;
  name: string;
  idApplicationType: number;
  applicationType: ApplicationType;

  constructor() {
    this.idApplication = 0;
    this.name = '';
    this.idApplicationType = 0;
    this.applicationType = new ApplicationType();
  }
}

File: EvalDI2P2P3API/Eval.Domain/Models/Application.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Eval.Domain.Models;

[Table("Application")]
public class Application
{
    [Key]
    public int IdApplication { get; set; }
    
    [Required]
    public required string Name { get; set; }
    
    [Required]
    public required int IdApplicationType { get; set; }
    
    [ForeignKey(nameof(IdApplicationType))]
    public ApplicationType? ApplicationType { get; set; }
}